### PR TITLE
Add Automated testing lab 

### PR DESCRIPTION
**Pull request by jamestupper**

### CODE REVIEW

File: docs/HOL_Automated-Testing/HOL_Automated-Testing.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    	assertNotNull(response);
+    	assertEquals(HttpStatus.CREATED, response.getStatusCode());
+    	
+    	ResponseEntity<List<CatalogItem>> newItems = controller.getCatalogItems();
+    	int sizeAfterAdds = newItems.getBody().size();
+    	
+    	assertEquals(sizeBeforeAdds + 2, sizeAfterAdds);
+	}
+
+The final result should look something like this:
```
Comment (dtzar) [inline docs/HOL_Automated-Testing/HOL_Automated-Testing.md]: Screenshot does not match adding "testAddMultipleCatalogItems()". Instead screenshow shows adding "testAddCatalogItems()".

File: docs/HOL_Automated-Testing/HOL_Automated-Testing.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+**2.** Lets go ahead and run these tests. In the menu select **Run -> Run As -> JUnit Test**.
+
+![](media/run_default_tests.png)
+
+This should open up the JUnit window and give you a summary of the test run.
+
+![](media/default_test_run.png)
+
+**3.** Paste the following code into the `CatalogControllerTest.java` file:
```
Comment (dtzar) [inline docs/HOL_Automated-Testing/HOL_Automated-Testing.md]: Please Clarify.  For instance, you are adding in the code at line 31..

File: docs/HOL_Automated-Testing/HOL_Automated-Testing.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+**Run the automated tests in an automated build:** You will then run the JUnit tests as part of the continuous integration build that was set up in the HOL Continuous Integration with PArts Unlimited MRP
+
+###1. Set up development environment###
+
+First, we need to set up Eclipse in order to work on the Parts Unlimited MRP App on our local machine. The Parts Unlimited MRP App uses Gradle ([https://gradle.org/](https://gradle.org/ "https://gradle.org/")), so we need to set up the code as Gradle projects in Eclipse.
+
+**1.** Open up **Eclipse**.
+
+**2.** Navigate to **[https://marketplace.eclipse.org/content/gradle-integration-eclipse-0#](https://marketplace.eclipse.org/content/gradle-integration-eclipse-0# "https://marketplace.eclipse.org/content/gradle-integration-eclipse-0#")** and drag the **Install** button onto your running **Eclipse** workspace.
```
Comment (dtzar) [inline docs/HOL_Automated-Testing/HOL_Automated-Testing.md]: I spent awhile trying to make the drag and drop work with Eclipse and finally gave up and just did a Help - Eclipse Marketplace, and searched for "gradle" and did the install from there.  Please reference this alternate location in txt.

File: docs/HOL_Automated-Testing/HOL_Automated-Testing.md (hunk 1 above)
Comment (jamestupper) [inline docs/HOL_Automated-Testing/HOL_Automated-Testing.md]: Copy/paste error. Same code, just named the method differently.

**Pull request merged**